### DISCUSSION TITLE
Make Vanna more friendly for non Jupyter application.

### DISCUSSION THREAD

**Discussion by yonitjio** [vanna-ai/vanna#367] [General]
I just started to try Vanna, and right up front,  I think it's not very friendly for anyone whose background is from regular python programming.

For example these two:
* PostgreSQL and psycopg2 supports many kind of setup for database connection (trusted, password, etc.) but Vanna forces using connection with the whole parameters filled (i.e., host, port, user, password, database).
* Vanna uses `print` instead of `logging`.

**Comment by zainhoda** [comment]
The methods are intended to be overridable and customizable. If you look at what any of the `vn.connect_to_...` method do, they are all just convenience methods that set the `vn.run_sql` method. You just need to provide any function that takes in SQL and returns a pandas dataframe. In the docs, if you choose "Other Database", you'll see how to set that up.
https://vanna.ai/docs/other-database-openai-vanna-vannadb/

The log function deliberately uses a print statement due to the behavior of Jupyter notebooks. You can override the log function with whatever logging you want to use:
https://github.com/vanna-ai/vanna/blob/main/src/vanna/base/base.py#L77-L78

Please also have a look at:
https://vanna.ai/docs/behavior-customization/
